src/public/listeningAlbum.js

```diff
@@ -0,0 +1,15 @@
+import io from 'socket.io-client';
```

Comment:
This isn't working for me; I'm seeing a browser error about this import:

![](https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/9094098/fb2613a2-395e-4039-a7d3-276382aeff19)

What we found worked previously for the orderNotifications page, was to import the socket-io client library in the HTML/EJS file:

`<script src="/socket.io/socket.io.js"></script>` (see: [https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/socket_pause_listening/src/views/orderNotifications.ejs#L7)](https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/socket_pause_listening/src/views/orderNotifications.ejs#L7))

Reply:
I had doubts about it as well, on this reason I added in eslintrc.json the following  (it is your first comment): 
 "parserOptions": {
    "ecmaVersion": 2023,
    "sourceType": "module" 
I tried it as an option to resolve the problem 😄
I have added in ejs as per your comment , commit 4dec20b67555d1faef548824f0a49ef22d8b2568
In this case I can remove parserOptions from eslintrc.json  , right?

Reply:
Yes I believe you can remove it.

In order to tell ESLint, that `io()` is an available global variable in this script, we should add the following comment at the top of the page: `/* global io */` (see similar comment in `orderNotifications.js` where we have both `io` and the `Toastify` library being imported as global scripts)

Reply:
I have removed from eslintrc.json ParserOptions and added global variable at the top listeningAlbum.js but I have it as a eslint problem anyway

<img width="964" alt="image" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/88336329/133d07d1-506a-43aa-922b-e0c65b5306b1">

Reply:
Yes we must remove this `import` line altogether: `import io from 'socket.io-client';`. It won't work because the js script won't be able to import without the parser options change; and even with the parser options change, that path to the socket client file is not valid.

It is not needed because we have already essentially 'imported' the socket script into the browser's global environment, in our HTML/EJS file here: https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/6274fe20437cfb68eb63fcfe1d10642b5eee86d9/src/views/listeningAlbum.ejs#L7

The guidance from the docs recommend doing it this way: https://socket.io/docs/v4/client-installation/#installation

Reply:
Thank you , it is removed  commit fd311ed43a88bf4c9fcdf64bfcf4c6b8ced38a99